RepoDb/RepoDb/Extensions/ExpressionExtension.cs

```diff
@@ -65,6 +65,22 @@ internal static bool IsMathematical(this Expression expression)
             return m_mathematicalExpressionTypes.Contains(expression.NodeType);
         }
 
+        #region GetField
+
+        /// <summary>
+        /// Gets the <see cref="Field"/> defined on the current instance of <see cref="BinaryExpression"/>
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <returns></returns>
+        public static Field GetField(this BinaryExpression expression)
+        {
+            var name = expression.GetName();
+            var type = expression.GetMemberType();
+            return new Field(name,type);
+        }
```

Comment:
Thank you for this.

My intention is to make this 2 calls a single call further :) - a part of memory optimization and performance :)

The codes at 77 and 78 is also 2 calls. Can we refactor this further to drill on the correct extended methods of Expression?

I am thinking that you also have to create the following.

     public static Field GetField(this UnaryExpression expression);
     public static Field GetField(this MethodCallExpression expression);
     public static Field GetField(this MemberExpression expression);

And might/could be inside the last static method, there you define the condition of only single execution. Like below?

            var member = expression.Member;
            if (member.IsPropertyInfo())
            {
                return member.ToPropertyInfo().AsField();
            }
            else if (member.IsFieldInfo())
            {
                return member.ToFieldInfo().AsField();
            }

Is it possible?

Reply:
ok, on it!

Reply:
@mikependon ive updated it, kindly check